tests/lib/nm/user_test.py

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import pytest
+
+from lib.compat import mock
+
+from libnmstate import nm
+
+
+@pytest.fixture
+def NM_mock():
+    with mock.patch.object(nm.user.nmclient, 'NM') as m:
+        yield m
+
+
+def test_create_setting_None(NM_mock):
```

Comment:
Naming of tests suggestion:
- test_create_setting_with_no_values
- test_create_setting_given_existing_config
- test_create_setting_given_non_existing_config

Reply:
It tests the case where no settings needs to be created, how about `test_create_no_setting`?

Reply:
sounds good, thanks.